File: test/solutions/171.jl
function titleToNumber(columnTitle::AbstractString)::Int64
    num = 0
    for i in 1:length(columnTitle)
        digit = columnTitle[length(columnTitle)-i+1]
        num += (Int(digit[1]) - 64) * 26^(i-1)
    end
    return num
end


@testset "titleToNumber tests" begin
    # Test case 1
    columnTitle = "A"
    expected_output = 1
    @test titleToNumber(columnTitle) == expected_output
    
    # Test case 2
    columnTitle = "AB"
    expected_output = 28
    @test titleToNumber(columnTitle) == expected_output
    
    # Test case 3
    columnTitle = "ZY"
    expected_output = 701
    @test titleToNumber(columnTitle) == expected_output
end

File: test/rust/tests/171.rs
// Using Horner's Rule
pub fn title_to_number(column_title: String) -> i32 {
    let base = 'A' as u32 - 1;
    let mut column_number = 0;

    for c in column_title.chars() {
        let digit = c as u32 - base;
        column_number = column_number * 26 + digit;
    }

    column_number as i32
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_example1() {
        let column_title = "A".to_string();
        assert_eq!(title_to_number(column_title), 1);
    }

    #[test]
    fn test_example2() {
        let column_title = "AB".to_string();
        assert_eq!(title_to_number(column_title), 28);
    }

    #[test]
    fn test_example3() {
        let column_title = "ZY".to_string();
        assert_eq!(title_to_number(column_title), 701);
    }
}
